drv.go

```diff
@@ -137,6 +137,174 @@ type connPool struct {
 	serverVersion VersionInfo
 }
 
+func (d *drv) openPool(P *PoolParams) (*connPool, error) {
+
+    // determine key to use for pool
+    poolKey := fmt.Sprintf("%s\t%s\t%s\t%d\t%d\t%d\t%s\t%s\t%s\t%t\t%t\t",
+            P.UserName, P.Password, P.DSN, P.MinSessions, P.MaxSessions,
+            P.SessionIncrement, P.WaitTimeout, P.MaxLifeTime, P.SessionTimeout,
+            P.Heterogeneous, P.EnableEvents)
+    fmt.Printf("Pool key: |%s|\n", poolKey)
+
+    // if pool already exists, return it immediately
+    if pool, ok := d.pools[poolKey]; ok {
```

Comment:
Serialize access with d.mu.Lock() and d.mu.Unlock():
d.mu.Lock()
pool, ok := d.pools[poolKey]
d.mu.Unlock()
if ok {

Reply:
Good point. I'll make that change.